OpenAI (Other) Connector PKI schema

antman1p opened the pull request:
## Summary

This pull request adds the `PkiOpenAi` provider to the `ConfigSchema` in `x-pack/plugins/stack_connectors/common/openai/schema.ts`, introducing new fields to config (`verificationMode`) and new fields to secret (`certificateData`, `privateKeyData`, and `caData`) to support future PKI-based authentication introduced in https://github.com/elastic/kibana/pull/219984. This is a schema-only change—no runtime behavior is modified yet—intended as an intermediate step for an upcoming release where the full PKI connector implementation will follow in a subsequent PR. The existing `OpenAi`, `AzureAi`, and `Other` provider schemas remain unchanged, preserving all current functionality.

As a community contributor, I’ve followed the advice from @pmuellr  Patrick Mueller to base this PR on `main` and split my work into two PRs: this schema-only change first, followed by the implementation. Since I don’t have automatic CI privileges, I’ll need an Elastic team member to trigger the CI job manually after review (e.g., with `@elasticmachine please run CI`).

No visual changes are involved, so no screenshots or gifs are included.


### Release Notes

**Label**: `release_note:skip`

As this is the schema PR, we will hold off on the enhancement release note until the new feature in https://github.com/elastic/kibana/pull/219984
---

### Additional Notes for Reviewers

- **Context**: Per Patrick Mueller’s guidance, this PR targets `main` as the first of two PRs for my PKI OpenAI connector feature. The second PR will build on this schema to add runtime logic and UI (e.g., a dropdown for `verificationMode`). I’ve split the work to ease review and integration into an intermediate release.
- **CI Request**: As a community contributor, I don’t have automatic CI runs. Please trigger CI after initial review with `@elasticmachine please run CI`.
- **Team**: I expect to work with the `gen-ai` team on this, with framework team approval (per Patrick). Please let me know who to tag for review!
- **Tests**: I haven’t added tests yet—should I include schema validation tests here, or wait for the implementation PR? Happy to adjust based on feedback.

<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

antman1p committed:
Add PkiOpenAi provider with verificationMode to OpenAI connector schema.ts

Add PkiOpenAi provider with verificationMode to OpenAI connector schema

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

pmuellr commented on x-pack/platform/plugins/shared/stack_connectors/common/openai/schema.ts:
```diff
@@ -3,6 +3,8 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
+ *
+ * PKI Funcitonality added by Antonio Piazza @antman1p
```
I'm not aware that we do attributions like this - we can always track who made what changes via git blame, and it's possible this may mess up future "change all the copyright statements" changes we occaisonally make :-)

pmuellr commented on x-pack/platform/plugins/shared/stack_connectors/common/openai/schema.ts:
```diff
@@ -34,6 +36,32 @@ export const ConfigSchema = schema.oneOf([
+    apiProvider: schema.oneOf([schema.literal(OpenAiProviderType.PkiOpenAi)]),
+    apiUrl: schema.string(),
+    defaultModel: schema.string({ defaultValue: DEFAULT_OPENAI_MODEL }),
+    certPath: schema.string({
```
I'm still a bit confused because AFAICT, this sort of config could only be used by a customer in an on-premise deployment - it can't be used in any cloud deployments because customers have no access to the file system.

As an example, for the action plugin config, we allow ca file names OR their data to be used: https://github.com/elastic/kibana/blob/ba49a0a83cd5aaaf5ed254eb4123e52048c2d6d9/x-pack/platform/plugins/shared/actions/server/config.ts#L47-L64

That will work in cloud ...

antman1p committed:
removed created by comment in schema.ts

antman1p committed:
Allowed for both on-prem and cloud configurations schema.ts

Renaming certPath to certificateFile and keyPath to privateKeyFile.
Adding certificateData and privateKeyData for raw content.
Supporting both single strings and arrays for file paths with minLength: 1 and minSize: 1.
Removing the .pem validation, deferring it to runtime.

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p commented:
Made changes suggested by @pmuellr

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p commented:
@pmuellr I just wanted to ping you ro see if there is any status update on this.

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

pmuellr commented:
Our team will review after the owners of the connector have reviewed.  I would contact James and tell him you're waiting for reviews from @elastic/security-generative-ai , @elastic/obs-ai-assistant and @elastic/appex-ai-infra

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic commented on x-pack/platform/plugins/shared/stack_connectors/common/openai/schema.ts:
```diff
@@ -34,6 +34,31 @@ export const ConfigSchema = schema.oneOf([
     headers: schema.maybe(schema.recordOf(schema.string(), schema.string())),
   }),
+  schema.object({
+    apiProvider: schema.oneOf([schema.literal(OpenAiProviderType.PkiOpenAi)]),
```
I think you'll need to update `OpenAiProviderType` to include `PkiOpenAi`

stephmilovic commented on x-pack/platform/plugins/shared/stack_connectors/common/openai/schema.ts:
```diff
@@ -34,6 +34,31 @@ export const ConfigSchema = schema.oneOf([
+      schema.literal('full'),
+      schema.literal('certificate'),
+      schema.literal('none'),
+    ], { defaultValue: 'full' }),
```
I'm wondering if this should default to `none` since there is currently no verification mode implemented?

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Expanded OpenAiProviderType.Other

Expanded OpenAiProviderType.Other and removed OpenAiProviderType.PkiOpenAi

antman1p committed:
Changed cert verification from "full" to "none"

Changed cert verification from "full" to "none"

antman1p commented:
I mode the changes suggested by @stephmilovic

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic commented:
@elasticmachine merge usptream

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic commented on x-pack/platform/plugins/shared/stack_connectors/common/openai/schema.ts:
```diff
@@ -32,6 +32,27 @@ export const ConfigSchema = schema.oneOf([
+        schema.literal('full'),
+        schema.literal('certificate'),
+        schema.literal('none'),
+      ], { defaultValue: 'none' })
```
I was wrong here, sorry about that!
```suggestion
      ], { defaultValue: 'full' })
```

From @pmuellr 

> Kibana security will eventually notice if the default is none :slightly_smiling_face: (they have in the past for some of our tests).  verificationMode should default to full, otherwise it's a considered a security issue.

antman1p committed:
Update x-pack/platform/plugins/shared/stack_connectors/common/openai/schema.ts

Changed back to "full"

Co-authored-by: Steph Milovic <stephanie.milovic@elastic.co>

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic commented:
buildkite test this

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/296780)
* Commit: 05e34cbf8cd95112051c613b8e6a8aa3e7433e97

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/296780#01967cf0-933e-4e5d-93bd-1f91ba35278e)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296780","number":296780,"commit":"05e34cbf8cd95112051c613b8e6a8aa3e7433e97"}],"number":296780}
buildkite-pr-comment-->

stephmilovic commented:
buildkite test this

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/296830)
* Commit: 8f338b1afd9d56ad48cda05c37c656c829360f5f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #296780](https://buildkite.com/elastic/kibana-pull-request/builds/296780) failed 05e34cbf8cd95112051c613b8e6a8aa3e7433e97

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296830","number":296830,"commit":"8f338b1afd9d56ad48cda05c37c656c829360f5f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296780","number":296780,"commit":"05e34cbf8cd95112051c613b8e6a8aa3e7433e97"}],"number":296830}
buildkite-pr-comment-->

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Added PKI Mode Toggle

Added PKI Mode Toggle to reveal options only when toggled on.

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

prodsecmachine commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/kibana/pr-checks/09edca28-4c70-4f32-8269-89ed7f8976c7?refs=6bcafab769fbcef0ccdcd5af6daff8f17623b6b5&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/kibana/pr-checks/09edca28-4c70-4f32-8269-89ed7f8976c7/license?refs=6bcafab769fbcef0ccdcd5af6daff8f17623b6b5&source=prChecksComment)

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Added CAFile and Data

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
aligned with implementation

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic committed:
schema changes > PKI to secrets

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic committed:
schema update

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

antman1p committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic commented:
buildkite test it

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/build_plugin_list_docs'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301776)
* Commit: 82b9937324a8eb84f72761a00dadefcc67828f7b

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/301776#0196f88a-8063-4baf-a6a7-b792fe05cd55)

### History
* :green_heart: [Build #296830](https://buildkite.com/elastic/kibana-pull-request/builds/296830) succeeded 8f338b1afd9d56ad48cda05c37c656c829360f5f
* :broken_heart: [Build #296780](https://buildkite.com/elastic/kibana-pull-request/builds/296780) failed 05e34cbf8cd95112051c613b8e6a8aa3e7433e97

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301776","number":301776,"commit":"82b9937324a8eb84f72761a00dadefcc67828f7b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296830","number":296830,"commit":"8f338b1afd9d56ad48cda05c37c656c829360f5f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296780","number":296780,"commit":"05e34cbf8cd95112051c613b8e6a8aa3e7433e97"}],"number":301776}
buildkite-pr-comment-->

stephmilovic commented:
buildkite test it

stephmilovic commented on docs/extend/plugin-list.md:
```diff
@@ -24,12 +24,12 @@ mapped_pages:
 | [dataViewFieldEditor](https://github.com/elastic/kibana/blob/main/src/platform/plugins/shared/data_view_field_editor/README.md) | The reusable field editor across Kibana! |
 | [dataViewManagement](https://github.com/elastic/kibana/blob/main/src/platform/plugins/shared/data_view_management) | WARNING: Missing or empty README. |
-| [dataViews](https://github.com/elastic/kibana/blob/main/src/platform/plugins/shared/data_views/README.mdx) | The data views API provides a consistent method of structuring and formatting documents and field lists across the various Kibana apps. It's typically used in conjunction with `SearchSource` for composing queries. |
+| [dataViews](https://github.com/elastic/kibana/blob/main/src/platform/plugins/shared/data_views/README.mdx) | The data views API provides a consistent method of structuring and formatting documents and field lists across the various Kibana apps. It's typically used in conjunction with <DocLink id="kibDevTutorialDataSearchAndSessions" section="high-level-search" text="SearchSource" /> for composi ...
```
I'm not sure about changes to this file, they were autocommitted by CI?

stephmilovic committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic commented:
buildkite test it

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301787)
* Commit: 8c511384c3cb82569f3a4b472201ab1f1163f663

### Failed CI Steps
* [AppEx AI-Infra Inference APIs FTR tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/301787#0196f8a1-c216-4937-b5f2-6e0c9c71d001)
* [AppEx AI-Infra Inference APIs FTR tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/301787#0196f8c0-feb0-4e76-b6a5-85b5243e09a2)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301787#0196f8c0-feb0-4e76-b6a5-85b5243e09a2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301787/jobs/0196f8c0-feb0-4e76-b6a5-85b5243e09a2/artifacts/0196f8c7-1f5a-45c5-9e0b-e758c7d5d7b3) AppEx AI-Infra Inference APIs FTR tests / Inference plugin - API integration tests Connector bedrock-claude-3-5-sonnet chatComplete API streaming disabled returns token counts
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301787#0196f8a1-c216-4937-b5f2-6e0c9c71d001) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301787/jobs/0196f8a1-c216-4937-b5f2-6e0c9c71d001/artifacts/0196f8c0-f07e-4c0b-a3af-d218d7fcdd63) AppEx AI-Infra Inference APIs FTR tests / Inference plugin - API integration tests Connector bedrock-claude-3-5-sonnet chatComplete API streaming enabled executes a tool when explicitly requested

### History
* :broken_heart: [Build #301776](https://buildkite.com/elastic/kibana-pull-request/builds/301776) failed 82b9937324a8eb84f72761a00dadefcc67828f7b
* :green_heart: [Build #296830](https://buildkite.com/elastic/kibana-pull-request/builds/296830) succeeded 8f338b1afd9d56ad48cda05c37c656c829360f5f
* :broken_heart: [Build #296780](https://buildkite.com/elastic/kibana-pull-request/builds/296780) failed 05e34cbf8cd95112051c613b8e6a8aa3e7433e97

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301787","number":301787,"commit":"8c511384c3cb82569f3a4b472201ab1f1163f663"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301776","number":301776,"commit":"82b9937324a8eb84f72761a00dadefcc67828f7b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296830","number":296830,"commit":"8f338b1afd9d56ad48cda05c37c656c829360f5f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296780","number":296780,"commit":"05e34cbf8cd95112051c613b8e6a8aa3e7433e97"}],"number":301787}
buildkite-pr-comment-->

stephmilovic committed:
fix secrets validation

stephmilovic commented:
buildkite test it

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301840)
* Commit: 0febb4bbebf0bd29cab3d87380da82f0589fa894

### Failed CI Steps
* [Jest Tests #5](https://buildkite.com/elastic/kibana-pull-request/builds/301840#0196f923-dd08-4a52-9c42-8ea71773ea80)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301840#0196f923-dd08-4a52-9c42-8ea71773ea80) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301840/jobs/0196f923-dd08-4a52-9c42-8ea71773ea80/artifacts/0196f947-a4d1-4f24-b95a-97d632debe4d) Jest Tests #<span></span>5 / HistoricalResults Super Date Picker when new date is selected should invoke fetchHistoricalResults with new start and end dates, from: 0 and remaining fetch query opts

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #301787](https://buildkite.com/elastic/kibana-pull-request/builds/301787) failed 8c511384c3cb82569f3a4b472201ab1f1163f663
* :broken_heart: [Build #301776](https://buildkite.com/elastic/kibana-pull-request/builds/301776) failed 82b9937324a8eb84f72761a00dadefcc67828f7b
* :green_heart: [Build #296830](https://buildkite.com/elastic/kibana-pull-request/builds/296830) succeeded 8f338b1afd9d56ad48cda05c37c656c829360f5f
* :broken_heart: [Build #296780](https://buildkite.com/elastic/kibana-pull-request/builds/296780) failed 05e34cbf8cd95112051c613b8e6a8aa3e7433e97

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301840","number":301840,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301787","number":301787,"commit":"8c511384c3cb82569f3a4b472201ab1f1163f663"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301776","number":301776,"commit":"82b9937324a8eb84f72761a00dadefcc67828f7b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296830","number":296830,"commit":"8f338b1afd9d56ad48cda05c37c656c829360f5f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296780","number":296780,"commit":"05e34cbf8cd95112051c613b8e6a8aa3e7433e97"}],"number":301840}
buildkite-pr-comment-->

stephmilovic commented on x-pack/platform/plugins/shared/stack_connectors/server/connector_types/openai/index.ts:
```diff
@@ -44,6 +48,17 @@ export const getConnectorType = (): SubActionConnectorType<Config, Secrets> => (
   renderParameterTemplates,
 });
 
+const secretsValidator = (secretsObject: Secrets) => {
```
Although this is schema only, the schema breaks an integration test unless we continue validating that empty secrets throw this `apiKey` error

pmuellr replied:
Ya, that's fine.  It's expected some changes like this will also need to be made with "schema-only" PRs ...

stephmilovic commented:
buildkite test it

stephmilovic commented:
buildkite test it

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/302124)
* Commit: 0febb4bbebf0bd29cab3d87380da82f0589fa894

### Failed CI Steps
* [AppEx AI-Infra Inference APIs FTR tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/302124#0196fe28-7f48-483a-99e9-18f27d255c28)
* [AppEx AI-Infra Inference APIs FTR tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/302124#0196fe49-c5d3-41dc-b9e6-d1b5f1d994d6)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/302124#0196fe28-7f48-483a-99e9-18f27d255c28) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/302124/jobs/0196fe28-7f48-483a-99e9-18f27d255c28/artifacts/0196fe49-b59e-4c51-b708-fcfb332e3bf0) AppEx AI-Infra Inference APIs FTR tests / Inference plugin - API integration tests Connector bedrock-claude-3-5-sonnet chatComplete API streaming disabled executes a tool with native function calling
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/302124#0196fe49-c5d3-41dc-b9e6-d1b5f1d994d6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/302124/jobs/0196fe49-c5d3-41dc-b9e6-d1b5f1d994d6/artifacts/0196fe50-fb7b-40fa-a1b2-df1f00c2b676) AppEx AI-Infra Inference APIs FTR tests / Inference plugin - API integration tests Connector bedrock-claude-3-5-sonnet chatComplete API streaming disabled returns a chat completion message for a simple prompt

### History
* :broken_heart: [Build #302099](https://buildkite.com/elastic/kibana-pull-request/builds/302099) failed 0febb4bbebf0bd29cab3d87380da82f0589fa894
* :yellow_heart: [Build #301840](https://buildkite.com/elastic/kibana-pull-request/builds/301840) was flaky 0febb4bbebf0bd29cab3d87380da82f0589fa894
* :broken_heart: [Build #301787](https://buildkite.com/elastic/kibana-pull-request/builds/301787) failed 8c511384c3cb82569f3a4b472201ab1f1163f663
* :broken_heart: [Build #301776](https://buildkite.com/elastic/kibana-pull-request/builds/301776) failed 82b9937324a8eb84f72761a00dadefcc67828f7b
* :green_heart: [Build #296830](https://buildkite.com/elastic/kibana-pull-request/builds/296830) succeeded 8f338b1afd9d56ad48cda05c37c656c829360f5f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302124","number":302124,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302099","number":302099,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301840","number":301840,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301787","number":301787,"commit":"8c511384c3cb82569f3a4b472201ab1f1163f663"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301776","number":301776,"commit":"82b9937324a8eb84f72761a00dadefcc67828f7b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/296830","number":296830,"commit":"8f338b1afd9d56ad48cda05c37c656c829360f5f"}],"number":302124}
buildkite-pr-comment-->

stephmilovic commented:
buildkite test it

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/302178)
* Commit: 0febb4bbebf0bd29cab3d87380da82f0589fa894

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #302124](https://buildkite.com/elastic/kibana-pull-request/builds/302124) failed 0febb4bbebf0bd29cab3d87380da82f0589fa894
* :broken_heart: [Build #302099](https://buildkite.com/elastic/kibana-pull-request/builds/302099) failed 0febb4bbebf0bd29cab3d87380da82f0589fa894
* :yellow_heart: [Build #301840](https://buildkite.com/elastic/kibana-pull-request/builds/301840) was flaky 0febb4bbebf0bd29cab3d87380da82f0589fa894
* :broken_heart: [Build #301787](https://buildkite.com/elastic/kibana-pull-request/builds/301787) failed 8c511384c3cb82569f3a4b472201ab1f1163f663
* :broken_heart: [Build #301776](https://buildkite.com/elastic/kibana-pull-request/builds/301776) failed 82b9937324a8eb84f72761a00dadefcc67828f7b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302178","number":302178,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302124","number":302124,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302099","number":302099,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301840","number":301840,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301787","number":301787,"commit":"8c511384c3cb82569f3a4b472201ab1f1163f663"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301776","number":301776,"commit":"82b9937324a8eb84f72761a00dadefcc67828f7b"}],"number":302178}
buildkite-pr-comment-->

pmuellr reviewed (approved):
LGTM

elasticmachine committed:
Merge branch 'main' into schema_openai_pki_main

stephmilovic reviewed (approved):
LGTM! Thanks @antman1p for your contribution and patience as we put this through the ringer!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/302720)
* Commit: 68856950a1212ebd4d8edc742e49fb2de9563b60

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #302178](https://buildkite.com/elastic/kibana-pull-request/builds/302178) succeeded 0febb4bbebf0bd29cab3d87380da82f0589fa894
* :broken_heart: [Build #302124](https://buildkite.com/elastic/kibana-pull-request/builds/302124) failed 0febb4bbebf0bd29cab3d87380da82f0589fa894
* :broken_heart: [Build #302099](https://buildkite.com/elastic/kibana-pull-request/builds/302099) failed 0febb4bbebf0bd29cab3d87380da82f0589fa894
* :yellow_heart: [Build #301840](https://buildkite.com/elastic/kibana-pull-request/builds/301840) was flaky 0febb4bbebf0bd29cab3d87380da82f0589fa894
* :broken_heart: [Build #301787](https://buildkite.com/elastic/kibana-pull-request/builds/301787) failed 8c511384c3cb82569f3a4b472201ab1f1163f663
* :broken_heart: [Build #301776](https://buildkite.com/elastic/kibana-pull-request/builds/301776) failed 82b9937324a8eb84f72761a00dadefcc67828f7b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302720","number":302720,"commit":"68856950a1212ebd4d8edc742e49fb2de9563b60"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302178","number":302178,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302124","number":302124,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302099","number":302099,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301840","number":301840,"commit":"0febb4bbebf0bd29cab3d87380da82f0589fa894"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301787","number":301787,"commit":"8c511384c3cb82569f3a4b472201ab1f1163f663"}],"number":302720}
buildkite-pr-comment-->

stephmilovic commented:
run docs-build

kibanamachine commented:
Starting backport for target branches: 8.19

https://github.com/elastic/kibana/actions/runs/15287006889

stephmilovic commented:
The implementation PR for this schema has been approved, so I merged the schema. I will wait to approve until next week, at which time I will merge the PR as part of our intermediary release process: https://github.com/elastic/kibana/pull/219984
